src/mechanics/timeIntegration/ExplicitRungeKutta.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <vector>
+
+namespace NuTo
+{
+namespace TimeIntegration
+{
+template <typename Tstate>
+class ExplicitRungeKutta
+{
+
+public:
+    //! @brief Initialization with method specific parameters (butcher tableau)
+    ExplicitRungeKutta(std::vector<std::vector<double>> aa, std::vector<double> bb, std::vector<double> cc)
+        : a(aa)
+        , b(bb)
+        , c(cc)
+    {
+    }
+
+    //! @brief Performs one RungeKutta step
+    //! @param f A functor that returns the right hand side of the differential equation
+    //!
+    //! The signature of its call operator must be:
+    //! operator()(const Tstate& w, Tstate& dwdt, double t)
+    //! The return value is stored in dwdt
+    //!
+    //! @param w0 initial value
+    //! @param t0 start time
+    //! @param h step size (t-t0)
+    //! @return value after one RungeKutta step
+    template <typename F>
+    Tstate DoStep(F f, Tstate w0, double t0, double h)
+    {
+        std::vector<Tstate> k(c.size(), w0);
```

Comment:
performance gain 1: make `c.size()` a constexpr (e.g. template argument `TNumStages`)
performance gain 2: `std::vector` --> `std::array<TState, TNumStages>`
The last one requires that `TState` is default constructable (IMO bad) OR a wierd template magic `make_array<TNumStages>(w0)` properly initializes the array with copies of `w0`. 

As long as performance is not important, this ugliness may not be worth it.